packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-processing-job.ts

```diff
@@ -0,0 +1,438 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Duration, Lazy, Size, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs'; import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import { AppSpecification, ExperimentConfig, NetworkConfig, ProcessingInput, ProcessingOutputConfig, ProcessingResources, StoppingCondition } from './base-types';
+import { renderTags } from './private/utils';
+
+/**
+ * Properties for creating an Amazon SageMaker processing job
+ *
+ * @experimental
+ */
+export interface SageMakerCreateProcessingJobProps extends sfn.TaskStateBaseProps {
+  /**
+   * Role for the Processing Job. The role must be granted all necessary permissions for the SageMaker processing job to
+   * be able to operate.
+   *
+   * See https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-createprocessingjob-perms
+   *
+   * @default - a role with appropriate permissions will be created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Configures the processing job to run a specified Docker container image.
+   */
+  readonly appSpecification: AppSpecification;
+
+  /**
+   * Sets the environment variables in the Docker container.
+   *
+   * @default - No environment variables
+   */
+  readonly environment?: { [key: string]: any };
+
+  /**
+   * Associates a SageMaker job as a trial component with an experiment and trial. Specified when you call the
+   * following APIs: CreateProcessingJob, CreateTrainingJob, CreateTransformJob
+   *
+   * @default - No experiment configuration
+   */
+  readonly experimentConfig?: ExperimentConfig;
+
+  /**
+   * Networking options for a processing job.
+   *
+   * @default - No network configuration
+   */
+  readonly networkConfig?: NetworkConfig;
+
+  /**
+   * For each input, data is downloaded from S3 into the processing container before the processing job begins running
+   * if 'S3InputMode' is set to File.
+   *
+   * @default - No processing inputs
+   */
+  readonly processingInputs?: ProcessingInput[];
+
+  /**
+   * The name of the processing job. The name must be unique within an AWS Region in the AWS account.
+   */
+  readonly processingJobName: string;
+
+  /**
+   * Output configuration for the processing job.
+   *
+   * @default - No processing output configuration
+   */
+  readonly processingOutputConfig?: ProcessingOutputConfig;
+
+  /**
+   * Identifies the resources, ML compute instances, and ML storage volumes to deploy for a processing job.
+   * In distributed training, you specify more than one instance.
+   *
+   * @default - 1 instance of EC2 `M4.XLarge` with `10GB` volume
+   */
+  readonly processingResources?: ProcessingResources;
```

Comment:
I can definitely see where you're coming from (I do agree that the `processing` prefix feels redundant here).

However, I'm hesitant to do as you suggested since I actually didn't come up with the names for these parameters. I'm modeling the CloudFormation API as closely as possible here.

As a couple of examples:
* https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ProcessingOutputConfig.html
* https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ProcessingResources.html

Although the code might feel `cleaner` if we do some or all of:
* Bundle these under their own type
* Rename them
* Some other abstraction

I think in the long run it will be harder for someone new to the module to make a contribution here if it doesn't match the source API names as closely as possible.

Thus, I'm thinking to keep these as they are. Let me know if you (or any other reviewer) feel strongly otherwise. This is my first contribution to CDK so I'm not aware of higher level patterns or general design tenants that we should try to be following.